Consumer/ConsumerManager.php

```diff
@@ -0,0 +1,148 @@
+<?php
+declare(strict_types = 1);
+
+namespace M6Web\Bundle\KafkaBundle\Consumer;
+
+use M6Web\Bundle\KafkaBundle\AbstractManager;
+
+/**
+ * Class Consumer
+ * @package M6Web\Bundle\KafkaBundle\Consumer
+ *
+ * A class to consume messages with topics set from the configuration or manually
+ */
+class ConsumerManager extends AbstractManager
+{
+    const ORIGIN = 'consumer';
+
+    /**
+     * @var \RdKafaka\Consumer
+     */
+    protected $entity;
+
+    /**
+     * @var \RdKafka\Topics[]
+     */
+    protected $topics = [];
+
+    /**
+     * @var \RdKafka\Queue
+     */
+    protected $queue;
+
+    /**
+     * @var int
+     */
+    protected $timeoutConsumingQueue;
+
+    /**
+     * @var TopicsConsumptionState
+     */
+    protected $topicsConsumptionState;
+
+    /**
+     * @return TopicsConsumptionState|null
+     */
+    public function getTopicsConsumptionState()
+    {
+        return clone $this->topicsConsumptionState;
+    }
+
+    /**
+     * @param TopicsConsumptionState $topicsConsumptionState
+     */
+    public function setTopicsConsumptionState(TopicsConsumptionState $topicsConsumptionState)
+    {
+        $this->topicsConsumptionState = $topicsConsumptionState;
+    }
+
+    /**
+     * @param TopicsConsumptionState $topicsConsumptionState
+     * @return ConsumerManager
+     * @throws \Exception
+     */
+    public function defineTopicsConsumptionState(TopicsConsumptionState $topicsConsumptionState): ConsumerManager
+    {
+        $this->checkIfEntitySet();
+        $this->checkIfBrokersSet();
+        $this->checkIfLogLevelSet();
+
+        $this->populateTopicsFromServer();
+        $topicsConsumptionState->setTopicsFromServer($this->topicsFromServer);
+
+        foreach ($this->topics as $topic) {
+            $topicsConsumptionState->defineTopic($topic->getName(), true);
+        }
+
+        $this->topicsConsumptionState = $topicsConsumptionState;
+
+        return $this;
+    }
+
+    /**
+     * @param int $timeoutConsumingQueue
+     *
+     * @return $this
+     */
+    public function setTimeoutConsumingQueue(int $timeoutConsumingQueue)
+    {
+        $this->timeoutConsumingQueue = $timeoutConsumingQueue;
+
+        return $this;
+    }
+
+    /**
+     * @return array
+     */
+    public function consumeStart()
+    {
+        if (empty($this->topicsConsumptionState)) {
+            throw new \Exception('No topics consumption state set');
+        }
+
+        $this->queue = $this->entity->newQueue();
+
+        $topics = $this->topicsConsumptionState->getTopics();
+        array_walk($topics, $this->setQueue());
+    }
+
+    /**
+     * @return array
+     */
+    public function consume(): \RdKafka\Message
```

Comment:
@lucascourot pour 

```php
    /**
     * @param int $logLevel
     * @return AbstractManager
     */
    public function setLogLevel(int $logLevel): AbstractManager
    {
        return $this;
    }
```

c'est justement la faute de PHP7 si il n'y a pas `@return $this` dans ce cas, car PhpStorm n'est pas content quand on a un `@return` différent du typage de retour php7... donc tant qu'on que php7 ne supporte pas `: $this` en typage de retour, on n'aura forcément une incohérence quelque part